README.md

```diff
@@ -50,16 +50,33 @@ For example, pennsieve.py file
 from sparc.client import SparcClient
 a = SparcClient(connect=False, config_file='config.ini')
 module = a.pennsieve.connect()
-module.user.whoami() #execute internal functions of the module
+module.info() #execute internal functions of the module
```

Comment:
Documentation with Sphinx is auto-generated, so this will be up to date. 
CI/Tests will show any potential issues.  
The only issue we need to manually update are tutorials and README.md, which I forgot to update - thanks for bringing this to my attention :smile:

Reply:
That is indeed my point: do the examples in the README.md are automatically tested? 
For instance, I personally like very much the approach in [pydantic doc](https://docs.pydantic.dev/usage/models/). All there examples are standalone scripts so they can test them by "extracting" them from the doc and running them

<img width="786" alt="image" src="https://user-images.githubusercontent.com/32402063/223683031-72370550-f19a-4748-9fff-fccf71e0f62a.png">

I am aware that in our case the environment is more complex than a "plain" library like pydantic (e.g. need probably authentication, ...) but I was just wondering if you had some thoughts on the matter :-)

Reply:
Pydantic looks interesting, but if we decide to adapt it, we may potentially run into some issues with code maintenance.

To give a context, pennsieve module has a generic **get()** and **post()** methods, which basically allow to query multiple endpoints on the Pennsieve server. Each json response may contain different fields, depending on which endpoint is queried, there could be even hundreds of such endpoints. And each API change on Pennsieve will result in having to update the pydantic code.  I wouldn't like to find in the situation that we would need to implement and later maintain multiple classes of **pydantic.BaseModel**, each one dedicated to a specific response, just for the reason to provide type validation for the code. 

To sum up, I do see some benefits of using pydantic, but not in the initial release. Converting to pydantic may require much more planning than anticipated. I think we can continue to have this discussion to have this feature, but I am closing this for now.

Reply:
I think you may have missed the point here @athril, my take from @pcrespov comment is are the code snippets contained within documentation tested?  I think probably [doctest](https://docs.python.org/3/library/doctest.html) is a solution for this concern.

Reply:
added #11 to make sure we don't forget this discussion and remind us to come back to this very useful capability - and we can continue the discussion on that issue.

Reply:
Oh, I understand better now. And doctest looks good to me too.
Thank you @hsorby for explaining.